Bindings/Java/OpenSimJNI/Test/testContext.cpp

```diff
@@ -223,21 +223,33 @@ int main()
     std::cout << pathAfterTypeChangeToViaInXML << endl;
  
     // Make a change to a socket that is invalid and verify that we can recover
-    // from that invalid change by restoring from a cached model and state.
+    // from that invalid change by not making it on model directly
     context->cacheModelAndState();
     Joint& shoulder = model->updComponent<Joint>("r_shoulder");
     AbstractSocket& socket = shoulder.updSocket("child_frame");
-    const std::string originalConnecteeName = socket.getConnecteeName();
     try {
         // create an invalid model where joint connects two frames on ground
-        socket.setConnecteeName("ground"); 
-        context->restoreStateFromCachedModel();
+        // this call leaves the model untouched if the change would invalidate topology
```

Comment:
@aseth1 there's a _model member variable which is the "live" model and there's a clonedModel used as a backup. Since these are member variables it makes no sense to pass them in. I can change the sentence to indicate _model instead if it helps

Reply:
*_model* and *clonedModel* are members of `OpenSimContext`, but here the local `model` is what we are attempting to edit. If it it is not the local *model* which owns the *socket* that is being modified then what is? Ideally, the comments would describe the condition that is being tested and its expected outcome. 

>// this call leaves the model untouched if the change would invalidate topology

How is that being verified in this tests?

Reply:
The constructor of OpenSimContext is passed in model and it holds to it as its member variable to mimic the GUI workflow. I agree this connection may not be clear but the goal of this test case is to replicate GUI workflow as close as possible. The statement is verified by not making any calls on the _model whatsoever!!